cli/picker/picker.go

```diff
@@ -0,0 +1,101 @@
+package picker
+
+import (
+	"bytes"
+	"os"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/cli/arg"
+	"github.com/buildbuddy-io/buildbuddy/cli/bazelisk"
+	"github.com/buildbuddy-io/buildbuddy/cli/log"
+	"github.com/manifoldco/promptui"
+)
+
+func HandlePicker(args []string) []string {
+	// If targets are already specified, don't do anything.
+	if len(arg.GetTargets(args)) > 0 {
+		return args
+	}
+
+	// If the command is build, test, or query without a specified target - apply to all targets.
+	command := arg.GetCommand(args)
+
+	// If it's a build, test, or query - apply to all targets.
+	if command == "build" || command == "test" || command == "query" {
+		args = append(args, "//...")
+		return args
+	}
+
+	// If it's not a run command, we're done here.
+	if command != "run" {
+		return args
+	}
+
+	// If it's a run, query executable targets.
+	queryArgs := []string{"query", `kind(.*_binary, //...) union kind(.*_application, //...) union attr(executable, 1, //...)`}
```

Comment:
I usually run all of my bazel commands from the root directory (honestly didn't even know you could run them from subdirectories lol)

Doesn't look like `:...` is a thing unless I'm doing something wrong:
```
❯ bazel query "kind(.*_binary, :...) union kind(.*_application, :...) union attr(executable, 1, :...)"
ERROR: no such target '//:...': target '...' not declared in package '' defined by /Users/siggi/Code/buildbuddy/BUILD (Tip: use `query "//:*"` to see all the targets in that package)
```

If you mean `:*` then that doesn't get subdirectories, which would break running from the root directory.

If you mean `...` without the `//` I get this if I try to run from a subdirectory (which I don't get from running at the root directory):
```
❯ bazel query "kind(.*_binary, ...) union kind(.*_application, ...) union attr(executable, 1, ...)"
ERROR: error loading package under directory '': error loading package 'buildbuddy': Unable to find package for @io_bazel_rules_docker//go:image.bzl: The repository '@io_bazel_rules_docker' could not be resolved: Repository '@io_bazel_rules_docker' is not defined.
```

Reply:
```
!master ~/work/buildbuddy/buildbuddy/cli> bazel query 'kind(.*_binary, ...) union kind(.*_application, ...) union attr(executable, 1, ...)'
//cli/cmd/bb:bb
//cli/cmd/bb:bb-darwin-amd64
//cli/cmd/bb:bb-darwin-arm64
//cli/cmd/bb:bb-linux-amd64
//cli/cmd/bb:bb-linux-arm64
//cli/cmd/bb:bb-windows-amd64
```

I think we might want to add `--keep-going` or `-k` to these query calls to make sure that we ignore irrelevant errors 🤔 .

I could make a PR tomorrow for this.

Reply:
Hm, that seems to be working for me now too - not sure what I was doing wrong. Maybe I had something else going on in my branch.

Reply:
Ah yeah, I see now - I had some other changes in my branch that were causing it to fail.